### PR TITLE
Native Input & Haptics (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by amydigiov**
This PR should not be merged or else EXR will break. These changes must first be merged into xr-authoring, and then EXR has to switch to using xr-authoring as a submodule, with input-prototype as a nested submodule.

See the PR in xr-authoring for a description of changes: [https://github.com/Unity-Technologies/xr-authoring/pull/2](https://github.com/Unity-Technologies/xr-authoring/pull/2)

### CODE REVIEW

File: FakePrototypeStuff/JoystickInputToEvents.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public void Update()
 	#region Non-Public Methods
 
 	// Fake gamepad has 10 axes (index 0 - 9) and 20 buttons (index 10 - 29).
```
Comment (AndrewTHEManeri) [inline FakePrototypeStuff/JoystickInputToEvents.cs:19]: Does the comment need updating?

File: Native/NativeInputDeviceManager.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine.Experimental.Input.Utilities;
+using UnityEngine.InputNew;
+using UnityEngineInternal.Input;
+
+namespace UnityEngine.Experimental.Input
+{
+    class NativeInputDeviceManager : ScriptableSettings<NativeInputDeviceManager>
```
Comment (AndrewTHEManeri) [inline Native/NativeInputDeviceManager.cs:9]: Does this really need to be a scriptablesettings?  It's bringing in a bunch of non-input code and it looks like discovered devices are added dynamically to the devicerecords anyways?

**Review by AndrewTHEManeri — CHANGES_REQUESTED** [review]
Was expecting to be using the 'old' input system axes and buttons and trackedpose driver, but after talking with Dalby we are just going to roll with this solution as it is temporary.

We'll need to put this in a separate branch for now since EXR isn't going to be switching to this right away.

Is the name change from Action to ControlHandle necessary?  That will provide some porting pain for EXR.

Also, anyway to do this without adding the scriptable settings to the prototype (Or moving that bit to the toolkit instead?)

File: Native/NativeInputDeviceManager.cs (hunk 2 above)
Comment (amydigiov) [reply Native/NativeInputDeviceManager.cs:9]: The native device manager needs to be a singleton and its records need to survive domain reload - ScriptableSettings seems like a good fit for that. Ideally we'd have some form of ScriptableSettings in Unity's API itself. But all that said, I can look into removing the dependency on non-input code.

**Comment by amydigiov** [comment]
@AndrewTHEManeri The name change to ControlHandle isn't necessary. I think it's a better name, and it makes the toolkit changes easier, but it's not necessary. I'll change it back - that's just a simple refactor->rename in my toolkit project.

**Pull request closed without merging**